components/apimgt/org.wso2.carbon.apimgt.rest.api/src/gen/java/org/wso2/carbon/apimgt/rest/api/ApisApi.java

```diff
@@ -275,5 +275,51 @@ public Response apisApiIdDocumentsDocumentIdDelete(@ApiParam(value = "**API ID**
     throws NotFoundException {
     return delegate.apisApiIdDocumentsDocumentIdDelete(apiId,documentId,ifMatch,ifUnmodifiedSince);
     }
+    @GET
+    @Path("/{apiId}/environments")
+    
+    
+    @io.swagger.annotations.ApiOperation(value = "", notes = "Get a list of documents belonging to an API.", response = Void.class)
+    @io.swagger.annotations.ApiResponses(value = { 
+        @io.swagger.annotations.ApiResponse(code = 200, message = "OK. Gateway environment list where this API is published returned."),
+        
+        @io.swagger.annotations.ApiResponse(code = 304, message = "Not Modified. Empty body because the client has already the latest version of the requested resource."),
+        
+        @io.swagger.annotations.ApiResponse(code = 404, message = "Not Found. Requested API does not exist."),
+        
+        @io.swagger.annotations.ApiResponse(code = 406, message = "Not Acceptable. The requested media type is not supported") })
+
+    public Response apisApiIdEnvironmentsGet(@ApiParam(value = "**API ID** consisting of the name of the API, the identifier of the version and of the provider of the API. \nShould be formatted as **name/version/provider**\n",required=true ) @PathParam("apiId") String apiId,
+    @ApiParam(value = "Maximum size of API array to return.",required=true) @QueryParam("limit") String limit,
+    @ApiParam(value = "Starting point of the item list.",required=true) @QueryParam("offset") String offset,
+    @ApiParam(value = "Search condition.") @QueryParam("query") String query,
+    @ApiParam(value = "Media types acceptable for the response. Should denote XML or JSON, default is JSON."  )@HeaderParam("Accept") String accept,
+    @ApiParam(value = "Validator for conditional requests; based on ETag."  )@HeaderParam("If-None-Match") String ifNoneMatch)
+    throws NotFoundException {
+    return delegate.apisApiIdEnvironmentsGet(apiId,limit,offset,query,accept,ifNoneMatch);
```

Comment:
Please note that these files get auto-generated using the swagger-to-cxf maven plugin we use for REST API development.